compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -422,6 +422,12 @@ impl<'a> Context<'a> {
         let (return_vars, return_warnings) =
             self.convert_ssa_return(entry_block.unwrap_terminator(), dfg)?;
 
+        let call_data_arrays: Vec<ValueId> =
+            self.data_bus.call_data.iter().map(|cd| cd.array_id).collect();
+        for call_data_array in call_data_arrays {
+            let _ = self.ensure_array_is_initialized(call_data_array, dfg);
```

Comment:
1. Does this `Result` need to be handled?
2. Could it be more convenient to construct the `Vec` of `call_data_arrays` and `ensure_array_is_initialized` in the same loop?

Reply:
1. call data is an Array it cannot fail, but it's better anyway to handle errors in case of. I fixed this.
2. It won't be more convenient because `ensure_array_is_initialized` mutate self.